rust/gitxet/tests/integration_tests/test_integration_scenarios.sh

```diff
@@ -0,0 +1,516 @@
+    ((intermediate_counter++))
+    >&2 new_repo ${tmp_repo}
+    echo ${tmp_repo}
+}
+
+
+name_conuter=0
+new_name() { 
+    base=$1
+    ((name_counter++))
+    echo "base_${name_counter}"
+}
+
+
+do_push() { 
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    pushd "$repo_dir_1"
+    git push --force "$repo_dir_2" 
+    popd
+}
+
+simulate_pr_merge() { 
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    local branch=${3:-main}
+    local remote_branch=${4:-main}
+
+    # Create a temporary repo from which to do the merge. 
+    tmp_repo=$(new_tmp_repo)
+    pushd "$tmp_repo"
+
+    # Disable git from running in the intermediate bit
+    git config --local filter.xet.process ""
+    git config --local filter.xet.required false
+
+    git remote add 1 "$repo_dir_1"
+    git remote add 2 "$repo_dir_2"
+
+    git fetch 1
+    git fetch 2
+
+    git reset --hard 1/$branch
+    git merge --no-edit 2/$remote_branch
+    git push 2
+    popd
+}
+
+
+# Create a new bare repo that simulates a fork of the original repo.
+simulate_fork() {
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    # Create a temporary new bare repo to push things to it.
+    local tmp_repo=$(new_tmp_bare_repo)
+    local tmp_repo_dir="$base_dir/$tmp_repo"
+
+    do_push $repo_1 $tmp_repo 
+
+    # Now go to the bare repo, which doesn't have the hooks or filter or anything, 
+    # and simply push from that to the new repo.  This should drop all the refs and 
+    # stuff.
+    new_bare_repo $repo_2
+
+    do_push $tmp_repo $repo_2 
+}
+
+
+
+# Scenario one.  Simple forking with a new repo.
+new_bare_repo origin_a 
+git clone origin_a repo_a_1
+
+# Add a text file, as if it's an existing repo 
+# Push the git change 
+pushd repo_a_1
+echo "I love this stuff" >> t1.txt
+git add t1.txt
+git commit -a -m "Added base text commit."
+git push origin main
+popd
+
+# Create some clones for a number of testing scenarios
+git clone origin_a repo_a_2
+git clone origin_a repo_a_3
+git clone origin_a repo_a_4
+git clone origin_a repo_a_5
+
+# Simulate a fork before the xet conversion happens 
+simulate_fork origin_a origin_b1
+simulate_fork origin_a origin_b2
+
+# Turn origin_a into a xet repo 
+init_repo_as_xet origin_a
+
+# Create some clones and forks in this scenario
+git clone origin_a repo_a_X1
+git clone origin_a repo_a_X2
+
+# Simulate a fork before the xet conversion happens 
+simulate_fork origin_a origin_x1
+simulate_fork origin_a origin_x2
+
+# Push another text file
+pushd repo_a_1
+git pull origin
+echo "I really love this stuff" >> t2.txt
+git add t2.txt
+git commit -a -m "Added another text commit."
+git push origin main
+popd
+
+# STATE:
+#  origin_a: XET, no data
```

Comment:
`no data` is a bit confusing. what about t1.txt and t2.txt?

Reply:
I could say no xet data.  I wanted to make sure there were some commits present that wouldn't get lost, since that was the main scenario.